src/ydata_synthetic/synthesizers/regular/cgan/model.py

```diff
@@ -102,6 +100,9 @@ def train(self, data: Union[DataFrame, array], label:str, train_arguments: Train
         self.data_dim = processed_data.shape[1] - len(self.processor.cat_pipeline.get_feature_names_out())
         self.define_gan()
 
+        mask = zeros(processed_data.shape[1], bool)  # Identify column indexes that carry the processed label feature
+        mask[[i for i, col in enumerate(list(self.processor._num_pipeline.get_feature_names_out()) + list(self.processor._cat_pipeline.get_feature_names_out())) if ''.join(col.split('_')[:-1]) == label_col]] = True
+
```

Comment:
After the whole data object is processed, this mask is created flagging the processed output columns that originate from label_col.